### PR TITLE
[663/docs]: Send data to Algolia and Show search bar and search resul…

### PR DESCRIPTION
**Pull request by herenacreative**
…t in the website

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Send data to Algolia before deploying the docs
2. Show search bar and search result in the website

## How did you implement / how did you fix it  
1.  install algoliasearch, @algolia/autocomplete-js, @algolia/autocomplete-theme-classic, react-instantsearch-dom
2. create user for api key in website algolia
3. change config apiKey in siteConfig
4. create search bar in file /SearchInput.js

## How to test  
1.  npm install algoliasearch react-instantsearch-dom
2. npm install @algolia/autocomplete-js 
3. npm install @algolia/autocomplete-theme-classic
5. search "old or rais" in search bar, then the search results appear

### CODE REVIEW

File: docs/src/siteConfig.js [hunk 1]
```diff
@@ -5,9 +5,10 @@ export const siteConfig = {
   npmUrl: 'https://www.npmjs.com/package/@hyperjumptech/monika',
   repoUrl: 'https://github.com/hyperjumptech/monika',
   algolia: {
-    appId: '',
-    apiKey: '',
-    indexName: '', // algoliaOptions: {
+    appId: '5O2YZGK4NG',
+    apiKey: '45f684bf6230e7980f10b4a9209fce83',
```
Comment (nicnocquee) [inline docs/src/siteConfig.js:9]: Don't hard code these. Read the value from the environment variable `ALGOLIA_APPLICATION_ID` and `ALGOLIA_SEARCH_API_KEY `

File: docs/src/siteConfig.js [hunk 2]
```diff
@@ -5,9 +5,10 @@ export const siteConfig = {
...
+    indexName: 'Foods',
```
Comment (nicnocquee) [inline docs/src/siteConfig.js:10]: use different name for the index, something related unlike `Foods`.

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: docs/src/components/SearchInput.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      .split(' ')
+      .map((token) => `^${token}`)
+      .join('|')})`,
+    flags
+  )
+  return pattern
+}
+
+function showMessage(message) {
+  console.log(message)
```
Comment (dennypradipta) [inline docs/src/components/SearchInput.js:30]: Is this necessary?

File: docs/src/components/SearchInput.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            setQuery('')
+          },
+        },
+        {
+          sourceId: options.indexName,
+          templates: {
+            item({ item, components }) {
+              return (
+                <a href={item.url}>
+                  <divc className="px-1">
```
Comment (haricnugraha) [inline docs/src/components/SearchInput.js:88]: is `divc` typo?

Comment (herenacreative) [reply docs/src/components/SearchInput.js]: yes typo, has been edited

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -37,3 +37,14 @@ jobs:
           branch: gh-pages # The branch the action should deploy to.
           folder: docs/out # The folder the action should deploy.
           clean: true # Automatically remove deleted files from the deploy branch
+
+      - name: Algolia
```
Comment (nicnocquee) [inline .github/workflows/docs.yml:41]: I think you need to call `npm run build-search` before running the Algolia step

File: .github/workflows/docs.yml [hunk 6]
```diff
@@ -37,3 +37,14 @@ jobs:
...
+        uses: wangchucheng/algolia-uploader@master
+        with:
+          app_id: ${{secrets.ALGOLIA_APPLICATION_ID}}# You can store token in your project's 'Setting > Secrets' and reference the name here. Such as ${{ secrets.ALGOLIA_ADMIN_KEY }}
+          admin_key: ${{secrets.ALGOLIA_SEARCH_API_KEY}}
+          # The index name. 
+          index_name: ${{secrets.ALGOLIA_INDEX_NAME}}
+          # The index file path relative to repo root.
+          index_file_path: ${{secrets.ALGOLIA_INDEX_FILE_PATH}}
```
Comment (nicnocquee) [inline .github/workflows/docs.yml:49]: There is no secret `ALGOLIA_INDEX_FILE_PATH `. You should put the path here directly.

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: docs/src/siteConfig.js (hunk 1 above)
Comment (herenacreative) [reply docs/src/siteConfig.js]: okay, i created the environment

File: docs/src/siteConfig.js (hunk 2 above)
Comment (herenacreative) [reply docs/src/siteConfig.js]: ok, renamed to monika-dokumentation

File: docs/src/components/SearchInput.js (hunk 3 above)
Comment (herenacreative) [reply docs/src/components/SearchInput.js]: changed to a components searchbox

File: .github/workflows/docs.yml (hunk 5 above)
Comment (herenacreative) [reply .github/workflows/docs.yml]: resolve

File: docs/src/pages/build-search.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
+const algoliasearch = require('algoliasearch')
```
[the remark this answers is not in the document]
Comment (herenacreative) [reply docs/src/pages/build-search.js]: ok, move to folder scripts

File: .github/workflows/docs.yml (hunk 6 above)
Comment (herenacreative) [reply .github/workflows/docs.yml]: resolve

**Review by nicnocquee — APPROVED** [review]

File: .github/workflows/docs.yml [hunk 8]
```diff
@@ -37,3 +37,12 @@ jobs:
...
           folder: docs/out # The folder the action should deploy.
           clean: true # Automatically remove deleted files from the deploy branch
+
+      - name: Generate and Save Algolia Object
+        uses: actions/setup-node@v2
+      - run: yarn
+      - run: yarn run buildsearch
+        env:
+          CI: true
+          ALGOLIA_APPLICATION_ID: ${{secrets.ALGOLIA_APPLICATION_ID}}
```
Comment (haricnugraha) [inline .github/workflows/docs.yml:47]: To make it consistent, please change this `ALGOLIA_APPLICATION_ID` to `NEXT_PUBLIC_ALGOLIA_APPLICATION_ID`

Comment (haricnugraha) [inline .github/workflows/docs.yml:47]: To make it consistent, please change this `ALGOLIA_APPLICATION_ID` to `NEXT_PUBLIC_ALGOLIA_APPLICATION_ID`

File: docs/scripts/build-search.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+ **********************************************************************************/
+
+const algoliasearch = require('algoliasearch')
+const fs = require('fs')
+const matter = require('gray-matter')
+const path = require('path')
+
+const options = {
+  indexName: 'monika-documentation',
+  appId: process.env.ALGOLIA_APPLICATION_ID || '5O2YZGK4NG',
```
Comment (haricnugraha) [inline docs/scripts/build-search.js:32]: To make it consistent, please change this `ALGOLIA_APPLICATION_ID` to `NEXT_PUBLIC_ALGOLIA_APPLICATION_ID`

File: docs/scripts/build-search.js [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+  if (!options.indexName) {
+    throw new Error('Algolia index name is not defined')
+  }
+
+  if (!options.appId) {
+    throw new Error('ALGOLIA_APPLICATION_ID is not defined')
+  }
+
+  if (!options.adminAPIKey) {
+    throw new Error('ALGOLIA_SEARCH_API_KEY is not defined')
```
Comment (haricnugraha) [inline docs/scripts/build-search.js:108]: `ALGOLIA_SEARCH_API_KEY` needs to change to `ALGOLIA_ADMIN_API_KEY`

Comment (haricnugraha) [inline docs/scripts/build-search.js:108]: `ALGOLIA_SEARCH_API_KEY` needs to change to `ALGOLIA_ADMIN_API_KEY`

**Review by nicnocquee — APPROVED** [review]

**Pull request merged**